### PR TITLE
Add `typeof(X)` scalar function

### PR DESCRIPTION
**Pull request by travenin**
See SQLite documentation: https://www.sqlite.org/lang_corefunc.html#typeof

Relates to issue #144 Scalar function support

### CODE REVIEW

File: core/vdbe/mod.rs [hunk 1]
```diff
@@ -1754,6 +1759,20 @@ fn exec_substring(
...
+    match reg {
+        OwnedValue::Null => OwnedValue::Text(Rc::new("null".to_string())),
+        OwnedValue::Integer(_) => OwnedValue::Text(Rc::new("integer".to_string())),
+        OwnedValue::Float(_) => OwnedValue::Text(Rc::new("real".to_string())),
+        OwnedValue::Text(_) => OwnedValue::Text(Rc::new("text".to_string())),
+        OwnedValue::Blob(_) => OwnedValue::Text(Rc::new("blob".to_string())),
+
+        // TODO: Unreachable?
+        OwnedValue::Record(_) => todo!(),
+        OwnedValue::Agg(_) => todo!(),
```
Comment (travenin) [inline core/vdbe/mod.rs:1772]: According to SQLite [`typeof` documentation](https://www.sqlite.org/lang_corefunc.html#typeof) the return value is either "null", "integer", "real", "text", or "blob". Maybe we should use `unreachable!()` macro here and believe that we never run function with this type of inputs.

File: testing/scalar-functions.test [hunk 2]
```diff
@@ -403,6 +403,38 @@ do_execsql_test date-with-invalid-timezone {
...
+  SELECT typeof(1.0);
+} {real}
+
+do_execsql_test typeof-blob {
+  SELECT typeof(x'61');
+} {blob}
+
+do_execsql_test typeof-blob-empty {
+  SELECT typeof(x'');
+} {blob}
```
Comment (travenin) [inline testing/scalar-functions.test:436]: Currently Limbo fails doesn't support blobs very well, and these blob tests fail with an error:

```
thread 'main' panicked at core/translate/expr.rs:1161:38:
not yet implemented
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
    while executing
"exec {*}$command"
    (procedure "evaluate_sql" line 3)
    invoked from within
"evaluate_sql $sqlite_exec $sql"
    (procedure "run_test" line 2)
    invoked from within
"run_test $::sqlite_exec $combined_sql $combined_expected_output"
    (procedure "do_execsql_test" line 5)
    invoked from within
"do_execsql_test typeof-blob {
  SELECT typeof(x'61');
} {blob}"
    (file "./testing/scalar-functions.test" line 430)
    invoked from within
"source $testdir/scalar-functions.test"
    (file "./testing/all.test" line 15)
make: *** [Makefile:59: test-compat] Error 1
```

File: core/vdbe/mod.rs (hunk 1 above)
Comment (jussisaurio) [inline core/vdbe/mod.rs:1772]: Aggregations at least should also be supported:

```
sqlite> select typeof(sum(age)) from users;
integer
sqlite> select typeof(group_concat(age)) from users;
text
```

Dunno if limbo plumbing in general supports it yet but worth a try

Comment (jussisaurio) [reply core/vdbe/mod.rs:1772]: @travenin limbo should now support e.g. `typeof(group_concat(age))` as well

Comment (travenin) [reply core/vdbe/mod.rs:1772]: Great to hear. I will craft this forward.

File: core/vdbe/mod.rs [hunk 3]
```diff
@@ -2132,6 +2134,18 @@ fn exec_substring(
...
 }
 
+fn exec_typeof(reg: &OwnedValue) -> OwnedValue {
+    match reg {
+        OwnedValue::Null => OwnedValue::Text(Rc::new("null".to_string())),
+        OwnedValue::Integer(_) => OwnedValue::Text(Rc::new("integer".to_string())),
+        OwnedValue::Float(_) => OwnedValue::Text(Rc::new("real".to_string())),
+        OwnedValue::Text(_) => OwnedValue::Text(Rc::new("text".to_string())),
+        OwnedValue::Blob(_) => OwnedValue::Text(Rc::new("blob".to_string())),
+        OwnedValue::Agg(ctx) => exec_typeof(ctx.final_value()),
```
Comment (travenin) [inline core/vdbe/mod.rs:2144]: I expect that this recursion causes no harm ever.

**Review by jussisaurio — APPROVED** [review]

**Pull request merged**